File: shared/src/main/java/xyz/klinker/messenger/shared/receiver/NotificationDismissedReceiver.java
package xyz.klinker.messenger.shared.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import xyz.klinker.messenger.shared.service.NotificationDismissedService;

public class NotificationDismissedReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        long conversationId = intent.getLongExtra(NotificationDismissedService.EXTRA_CONVERSATION_ID, 0L);

        Intent serviceIntent = new Intent(context, NotificationDismissedService.class);
        serviceIntent.putExtra(NotificationDismissedService.EXTRA_CONVERSATION_ID, conversationId);

        context.startService(serviceIntent);
    }
}

File: shared/src/main/java/xyz/klinker/messenger/shared/receiver/NotificationDismissedReceiver.kt
package xyz.klinker.messenger.shared.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

import xyz.klinker.messenger.shared.service.NotificationDismissedService

class NotificationDismissedReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        val conversationId = intent.getLongExtra(NotificationDismissedService.EXTRA_CONVERSATION_ID, 0L)

        val serviceIntent = Intent(context, NotificationDismissedService::class.java)
        serviceIntent.putExtra(NotificationDismissedService.EXTRA_CONVERSATION_ID, conversationId)

        Thread { NotificationDismissedService.handle(serviceIntent, context) }.start()
    }
}
